projects/elements/src/lib/lazy-elements/lazy-elements-loader.service.ts

```diff
@@ -2,21 +2,35 @@ import { Injectable } from '@angular/core';
 
 const LOG_PREFIX = '@angular-extensions/elements';
 
+export interface ElementConfig {
+  tagName: string;
+  url: string;
+}
+
 @Injectable({
   providedIn: 'root'
 })
 export class LazyElementsLoaderService {
   registry: Map<string, Promise<void>> = new Map<string, Promise<void>>();
+  configs: ElementConfig[] = [];
 
   constructor() {}
 
+  addConfig(configs) {
+    this.configs = [...this.configs, ...configs];
+  }
+
   loadElement(
     url: string,
     tag: string,
     isModule: boolean = false
   ): Promise<void> {
     if (!url) {
-      throw new Error(`${LOG_PREFIX} - url for <${tag}> not found`);
+      const config = this.configs.find(c => c.tagName === tag);
```

Comment:
Let's separate it and give it names so the `const elementUrl` can either be `url` passed from outside or `config.url` so that we do not re-assign the original `url`